core/src/main/java/com/graphhopper/routing/util/CarFlagEncoder.java

```diff
@@ -37,13 +41,12 @@
 public class CarFlagEncoder extends AbstractFlagEncoder {
     protected final Map<String, Integer> trackTypeSpeedMap = new HashMap<>();
     protected final Set<String> badSurfaceSpeedMap = new HashSet<>();
+    protected final Set<String> avoidAccessAtAllCosts = new HashSet<>();
```

Comment:
ok but I still do not understand why it is called `avoidAccessAtAllCosts` because we do not want to 'avoid access at all costs'. We want to 'apply penalty so normally we do not pass these roads unless its required to arrive at the destination (or start)'.

Reply:
Sorry, I do not understand the difference why avoiding a road isn't the same as adding a penalty to it. If a road is only accessible for people living there then I want to avoid the access of that road under all circumstances for the general path finding.

Reply:
Hm for me 'avoiding something at all costs' sounds like I will not do it under any circumstances and 'applying a penalty' is more like I rather do not do it unless its really needed and I accept some penalty for doing it :) 

> If a road is only accessible for people living there then I want to avoid the access of that road under all circumstances for the general path finding.

Yes and no :) I want to avoid that road under all circumstances **when I am only passing through**, but I still use it if I need to use it to arrive at the destination. And even for passing through I do not avoid it under all circumstances, because the penalty is not infinite, i.e. if the required detour is too large I still use it.